### PR TITLE
Fix ThirdPartyVersionsRule version check

### PR DESCRIPTION
**Pull request by ajaybhargavb**
#319 

@pakrym @Eilon @pranavkm 

I ran this on the latest packages on both 2.0.0 and dev and it passes.

### CODE REVIEW

**Comment by ajaybhargavb** [comment]
🆙 📅

**Comment by natemcmaster** [comment]
Can we add some unit tests to help us catch this in the future?

File: src/NuGetPackageVerifier/Rules/ThirdPartyPackageVersionsRule.cs [hunk 1]
```diff
@@ -55,7 +55,15 @@ private PackageVerifierIssue VerifyPackageReference(IPackageMetadata contextMeta
             foreach (var expectedVersion in expectedVersions)
             {
                 var expectedSemVersion = VersionRange.Parse(expectedVersion);
-                if (packageVersion.IsSubSetOrEqualTo(expectedSemVersion))
+                if (expectedSemVersion.IsFloating)
+                {
+                    if (expectedSemVersion.Float.Satisfies(packageVersion.MinVersion) &&
+                        expectedSemVersion.MaxVersion == packageVersion.MaxVersion)
```
Comment (pranavkm) [inline src/NuGetPackageVerifier/Rules/ThirdPartyPackageVersionsRule.cs]: `Equals`?

Comment (ajaybhargavb) [inline src/NuGetPackageVerifier/Rules/ThirdPartyPackageVersionsRule.cs]: I decided not to because this handles null case as well. They are `IEquatable`.

**Comment by ajaybhargavb** [comment]
> Can we add some unit tests to help us catch this in the future?

@natemcmaster, is it okay if I file an issue for this? We need a broader test coverage anyway. We don't have any tests for NPV.

**Comment by natemcmaster** [comment]
Yes, a follow up item is fine for now.

**Comment by natemcmaster** [comment]
:shipit:

**Comment by ajaybhargavb** [comment]
Filed https://github.com/aspnet/BuildTools/issues/329

**Comment by Eilon** [comment]
Talked to @ajaybhargavb in person, approved for 2.0.

**Pull request merged**